ui/src/layout/ActivityPanel.js

```diff
@@ -81,7 +87,27 @@ const ActivityPanel = () => {
           dispatch(scanStatusUpdate(data.scanStatus))
         }
       })
-  }, [dispatch])
+    if (open) {
```

Comment:
the `useEffect` runs whenever open changes it state where `open = Boolean(anchorEl)` i.e whenever we either open/close the modal but we don't want to make an API call when we close the modal hence this check condtion.